src/main/scala/mesosphere/marathon/core/instance/GoalAdjustmentReason.scala

```diff
@@ -0,0 +1,30 @@
+package mesosphere.marathon
+package core.instance
+
+/**
+  * Enumeration for the reason why an instance's goal has been adjusted.
+  */
+sealed trait GoalAdjustmentReason extends Product with Serializable
+
+object GoalAdjustmentReason {
+  /** The instance is killed because the instance count is scaled down */
+  case object OverCapacity extends GoalAdjustmentReason
+
+  /** Same as [[mesosphere.marathon.core.instance.GoalAdjustmentReason.OverCapacity]] but during deployment */
+  case object DeploymentScaling extends GoalAdjustmentReason
+
+  /** The instance is killed because the app has is being deleted */
+  case object DeletingApp extends GoalAdjustmentReason
+
+  /** The instance is decommissioned because of an incoming http request */
+  case object UserRequest extends GoalAdjustmentReason
+
+  /** The instance is killed because a new version is being deployed */
+  case object Upgrading extends GoalAdjustmentReason
+
+  /** The instance is killed because the app no longer exists */
+  case object Orphaned extends GoalAdjustmentReason
+
+  /** The instance is killed because it didn't turn running within a given time frame */
+  case object Overdue extends GoalAdjustmentReason
+}
```

Comment:
Hm, it feels we are not coherent. Eg `Upgrading` vs `DeletingApp`. So it should be `UpgradingInstance` and `DeletingApp`. We also mix verbs and adjectives. We probably should use only words that describe a current state or words that describe a current action, not both.

Reply:
So we would do `Orphaned` -> `RemovingApp` but e.g. `Overdue`?

Reply:
I just saw that we always used these names. So it should rather be `OrphanedInstance` and `UpgradingApp`. Or `InstanceOrphanRemoval` and `AppUpgrade`. I guess we could discuss this for ages so let's leave it for now.